tikv/raftstore/keys.go

```diff
@@ -101,10 +103,41 @@ func RegionStateKey(regionID uint64) []byte {
 	return key
 }
 
+func ValidateDataKey(key []byte) bool {
+	return len(key) > 0 && key[0] == DataPrefix
+}
+
+func DataKey(key []byte) []byte {
+	v := make([]byte, 0, 1+len(key))
```

Comment:
Is it possible to reuse memory space with key in golang?

Reply:
Yes, just pass a []byte as an argument then appends to it.

we can optimize it later.